VideoWeb/VideoWeb/ClientApp/src/app/shared/footer/footer.component.ts

```diff
@@ -1,38 +1,39 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { NavigationEnd, Router, RouterEvent } from '@angular/router';
 import { TranslateService } from '@ngx-translate/core';
 import { filter } from 'rxjs/operators';
 import { Logger } from 'src/app/services/logging/logger-base';
 import { pageUrls } from '../page-url.constants';
-
+import { Subscription } from 'rxjs';
 @Component({
     selector: 'app-footer',
     templateUrl: './footer.component.html',
     styleUrls: ['./footer.component.css']
 })
-export class FooterComponent implements OnInit {
+export class FooterComponent implements OnInit, OnDestroy {
     hideContactUsLink = false;
     privacyPolicyUri = pageUrls.PrivacyPolicy;
     accessibilityUri = pageUrls.Accessibility;
-    hideLinksForUnsupportedBrowser = false;
+    routerEventsSubscription$: Subscription = new Subscription();
 
     constructor(private router: Router, private translate: TranslateService, private logger: Logger) {
-        this.router.events.pipe(filter((event: RouterEvent) => event instanceof NavigationEnd)).subscribe(x => {
-            this.hideContactUs();
-            this.hideLinks();
-        });
+        this.routerEventsSubscription$.add(
+            this.router.events.pipe(filter((event: RouterEvent) => event instanceof NavigationEnd)).subscribe(x => {
+                this.hideContactUs();
+            })
+        );
     }
 
     ngOnInit() {
         this.hideContactUs();
     }
 
-    hideContactUs() {
-        this.hideContactUsLink = this.router.url === '/contact-us';
+    ngOnDestroy(): void {
+        this.routerEventsSubscription$.unsubscribe();
```

Comment:
This will work but we should probably settle on a pattern. I like the takeUntil onDestroy pattern @robertbennett1998  has used in some of the changes on the waiting room components. We can also look at other solutions such as this https://github.com/ngneat/until-destroy

JohWaitingRoomComponent
JudgeWaitingRoomComponent
ParticipantWaitingRoomComponent

Reply:
I agree we need to make this consistent and I think the takeUnitl approach or the package should be used as it is more explicit.